### DISCUSSION TITLE
Altair chart, exclude vertical marks from selection

### DISCUSSION THREAD

**Discussion by jakub-mikians-airspace-intelligence** [marimo-team/marimo#2689] [Q&A]
Hi

First, let me praise marimo contributors for the awesome job they do!

I have a chart, where I display `my_dataframe` as vertical bars. I also want to add a single vertical line to the chart. What's below is a fragment of the original code that fails (it's not the complete example)

```python
# This is not a complete example.

bars = (
    alt.Chart(my_dataframe)
    .mark_bar(size=bar_width)
    .encode(
        x=alt.X(
            "my_datetime:T",
            ...
        ),
        y="sum(count):Q",
        color=alt.Color("status:N", scale=status_colors),
    )
)

vertical_line = rules = (
    alt.Chart(pd.DataFrame({"vertical_line": [vertical_line_time], "color": ["black"]}))
    .mark_rule()
    .encode(x="vertical_line:T", color=alt.Color("color:N", scale=None))
)

final_chart = bars + vertical_line

final_chart
chart = mo.ui.altair_chart(
    final_chart,
    chart_selection=True,
    legend_selection=True,
)
chart
```

The bars display correctly and the vertical mark displays correctly as well. Now, when I select the bar and want to filter a table based on selection, the code does not work because the selection contains `vertical_line` value.

```python
pd.set_option("display.max_rows", None)
# The code below fails.
# chart.selections contians `vertical_line`, but my dataframe does not contain such a field.
table = mo.ui.table(
    chart.apply_selection(
        df_flights[
            [
                "my_datetime",
                "my_other_value",
            ]
        ]
    )
)
table
```

I don't want the selection to contain `vertical_line` value. How to combine chart with a vertical line and make it work with selection at the same time?

**Comment by mscolnick** [comment]
Maybe we can always emit `mark_rule` from selection. I am not sure if there is a good reason for that to be included (at least by default).

Without making changes to marimo, I am not quite sure if there is a way around that.
